packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js

```diff
@@ -44,7 +46,14 @@ const LocalePicker = () => {
   const dispatch = useDispatch();
   const pluginOptions = useSelector(selectContentManagerListViewPluginOptions);
   const locales = useSelector(selectI18NLocales);
+  const colllectionTypesRelatedPermissions = useSelector(selectCollectionTypesRelatedPermissions);
   const [{ query }, setQuery] = useQueryParams();
+  const {
```

Comment:
Can you create a custom hook somewhere for this? Something like:

```js
const locales = useAllowedLocales()
```

please?

Reply:
yes!

Reply:
I could not create a generic hook for the displayed locales as the logic is a bit different in the edit view. However I have extracted a hook that returns the needed permissions.

Reply:
what's different in the edit view?